sos/plugins/docker.py

```diff
@@ -29,20 +29,47 @@ class Docker(Plugin):
         ("all", "enable capture for all containers, even containers "
             "that have terminated", 'fast', False),
         ("logs", "capture logs for running containers",
-            'fast', False)
+            'fast', False),
+        ("size", "capture image sizes for ps", 'slow', False)
     ]
 
     def setup(self):
         self.add_copy_spec([
             "/var/lib/docker/repositories-*"
         ])
 
-        for subcmd in ['info', 'ps', 'ps -a', 'images', 'version']:
+        for subcmd in [
+            'info',
+            'images',
+            'network ls',
+            'ps',
+            'ps -a',
+            'stats --no-stream',
+            'version'
+        ]:
             self.add_cmd_output(
                 "{0} {1}".format(self.docker_cmd, subcmd)
             )
 
+        # separately grab ps -s as this can take a *very* long time
+        if self.get_option('size'):
+            self.add_cmd_output('{0} ps -as'.format(self.docker_cmd))
+
         self.add_journal(units="docker")
+        self.add_cmd_output("ls -alhR /etc/docker")
+
+        net_cmd = '{0} network ls'.format(self.docker_cmd)
+        nets = self.get_command_output(net_cmd)
+
+        if nets['status'] == 0:
+            n = [n.split()[1] for n in nets['output'].splitlines()[1:]]
+            for net in n:
+                self.add_cmd_output(
+                    "{0} network inspect {1}".format(
+                        self.docker_cmd,
+                        net
+                    )
+                )
```

Comment:
What's this getting? I don't think it matters too much for now, but some of this is beginning to look like some of the extensions we've made to `networking` in the past few years - at some point we will likely need to take a step back and consolidate that across plugins so it would be good to document (roughly) in the comments.